test/integration/zika.test.js

```diff
@@ -1,20 +1,92 @@
+import { toMatchImageSnapshot } from './helpers';
+
 describe("Zika", () => {
-  it('displays the title on the page', async () => {
-    await goToZikaPage()
+  it("displays the title on the page", async () => {
+    await goToZikaPage();
     await expect(page).toMatch("Real-time tracking of Zika virus evolution");
   });
 
-  it('clicks `Play` on the `TransmissionsCard`', async () => {
-    await goToZikaPage()
+  it("clicks `Play` on the `TransmissionsCard`", async () => {
+    await goToZikaPage();
+
+    await expect(page).toClick("button", { text: "Play" });
+
+    await expect(page).toMatchElement("button", { text: "Pause" });
+  });
+
+  describe("Color by", () => {
+    describe("author", () => {
+      it("matches the screenshot", async () => {
+        await matchSelectOptionScreenshot("author", async () => {
+          await page.keyboard.press("ArrowUp");
+          await page.keyboard.press("Enter");
+        });
+      });
+    });
 
-    await expect(page).toMatchElement('#TransmissionsCard')
+    describe("country", () => {
+      it("matches the screenshot", async () => {
+        await matchSelectOptionScreenshot("country", () => {});
+      });
+    });
 
-    await expect(page).toClick('button', { text: 'Play' })
+    describe("date", () => {
+      it("matches the screenshot", async () => {
+        await matchSelectOptionScreenshot("date", async () => {
+          await page.keyboard.press("ArrowUp");
+          await page.keyboard.press("ArrowUp");
+          await page.keyboard.press("Enter");
+        });
+      });
+    });
 
-    await expect(page).toMatchElement('button', { text: 'Pause' })
-  })
+    describe("region", () => {
+      it("matches the screenshot", async () => {
+        await matchSelectOptionScreenshot("region", async () => {
+          await page.keyboard.press("ArrowDown");
+          await page.keyboard.press("Enter");
+        });
+      });
+    });
+  });
 });
 
+async function matchSelectOptionScreenshot(option, selectOptionTest) {
+  await goToZikaPage();
+
+  const colorBySelector = await expect(page).toMatchElement("#selectColorBy");
+
+  await colorBySelector.click();
+
+  await selectOptionTest();
+
+  await expect(colorBySelector).toMatch(option);
+
+  const treeTitle = await expect(page).toMatchElement("#Title");
+
+  await expect(treeTitle).toMatch(option);
+
+  await toMatchImageSnapshot(async () => {
+    const image = await page.screenshot();
+
+    /**
+     * (tihuan): Apply `blur` and `failureThreshold` to ignore minor noises.
+     * Also `customSnapshotIdentifier` is needed, since we use `jest.retryTimes()`
+     * https://github.com/americanexpress/jest-image-snapshot/pull/122/files
+     * https://github.com/americanexpress/jest-image-snapshot#%EF%B8%8F-api
+     */
+    const SNAPSHOT_CONFIG = {
+      failureThreshold: 5,
```

Comment:
similar to my prior comment, I was finding I was able to cause quite a few visual oddities in the page, before I triggered an error with the 5% threshold. To decide on the `failureThreshold` and `blur` values, I think a good approach might be to:

1. purposely remove a couple elements from the page that aren't tested (make sure the missing elements trigger a failure in a reasonable amount of time).
2. muck with CSS to make the page look noticeably visually different, and make sure we get a failure in a reasonable amount of time 👌